providers/fips/selftest.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/params.h>
+#include <openssl/core_numbers.h>
+
+typedef struct self_test_post_params_st {
+    /* FIPS module integrity check parameters */
+    char *module_filename;                /* Module file to perform MAC on */
+    unsigned char *module_checksum_data;  /* Expected module MAC integrity */
+    size_t module_checksum_len;
```

Comment:
How is this len populated (and `indicator_checksum_len` below)?

Reply:
Will remove those variables for now and make the data all const char * (since they are all UTF8_ptr). When I add the code that accesses them it will need to convert some of the strings into unsigned char */ len